Cannot use webgme-taxonomy as an npm package

Cannot install the taxonomy as part of a bigger project (check webgme-cwl for reference)...
The most obvious issue is the internal 'modules' inside taxonomy. That should be possible to fix by simply using these subdirectories as npm packages (check the 'install-fix'), however apparently there are other weird behaviors as well:
- json importer usage and directly manipulating the requirejs pathes from code breaks the config:

Error: ENOENT: no such file or directory, scandir 'C:\WORK\Repos\webgme-cwl\node_modules\webgme-taxonomy\config/../node_modules/webgme-json-importer/src/plugins'
    at Object.readdirSync (node:fs:1451:3)
    at Object.getComponentNames (C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src\utils.js:129:25)
    at addAssetFromBasePath (C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js:83:42)
    at Object.addToRequireJsPaths (C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js:128:5)
    at initialize (C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src\server\worker\simpleworker.js:46:16)
    at process.<anonymous> (C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src\server\worker\simpleworker.js:70:16)
    at process.emit (node:events:513:28)
    at emit (node:internal/child_process:937:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  syscall: 'scandir',
  code: 'ENOENT',
  path: 'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme-taxonomy\\config/../node_modules/webgme-json-importer/src/plugins'
}

Please handle it with priority as it makes extremely difficult to test and create new deployments.
I believe having an npm package would be just a partial solution and this should work without that...

I am sorry to report, but it still appears that something is off:
'''
PS C:\WORK\Repos\webgme-cwl> node .\app.js
{
  enable: true,
  allowDuplication: true,
  defaultProject: 'cwl_base',
  basePaths: [
    'C:\\WORK\\Repos\\webgme-cwl\\config/../src/seeds/CWL',
    'C:\\WORK\\Repos\\webgme-cwl\\config/../node_modules/webgme-taxonomy/src/seeds/taxonomy'
  ],
  createAtStartup: []
}
2023-03-09T03:05:18.901Z - info: [gme] Node version v18.14.2
2023-03-09T03:05:18.903Z - info: [gme] Logfile [ info ] : C:\WORK\Repos\webgme-cwl\server.log
2023-03-09T03:05:18.903Z - info: [gme] Logfile [ error ] : C:\WORK\Repos\webgme-cwl\server-error.log
2023-03-09T03:05:18.921Z - info: [gme:server:standalone] Valid addresses of gme web server:  http://192.168.68.77:8888  http://127.0.0.1:8888  http://172.18.32.1:8888
Error: Tried loading "webgme-json-importer/build/JSONImporter.umd" at C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/build/JSONImporter.umd.js then tried node's require("C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd") and it failed with error: Error: Cannot find module 'C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd'
Require stack:
- C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js
- C:\WORK\Repos\webgme-cwl\node_modules\webgme\webgme.js
- C:\WORK\Repos\webgme-cwl\app.js
    at C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2600:27
    at Object.execCb (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1946:33)
    at Module.check (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1133:51)
    at Module.enable (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1426:22)
    at Module.init (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1038:26)
    at callGetModule (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1453:63)
    at Object.completeLoad (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1840:21)
    at req.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2613:17)
    at Object.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1935:21)
    at Module.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1084:29) {
  originalError: Error: Cannot find module 'C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd'
  Require stack:
  - C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js
  - C:\WORK\Repos\webgme-cwl\node_modules\webgme\webgme.js
  - C:\WORK\Repos\webgme-cwl\app.js
      at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
      at Module._load (node:internal/modules/cjs/loader:920:27)
      at Module.require (node:internal/modules/cjs/loader:1141:19)
      at require (node:internal/modules/cjs/helpers:110:18)
      at C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2598:75
      at Object.execCb (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1946:33)
      at Module.check (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1133:51)
      at Module.enable (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1426:22)
      at Module.init (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1038:26)
      at callGetModule (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1453:63) {
    code: 'MODULE_NOT_FOUND',
    requireStack: [
      'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme-engine\\index.js',
      'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme\\webgme.js',
      'C:\\WORK\\Repos\\webgme-cwl\\app.js'
    ]
  },
  moduleName: 'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme-engine\\src/..\\..\\webgme-json-importer\\src\\common/./build/JSONImporter.umd',
  requireModules: [ 'webgme-json-importer/build/JSONImporter.umd' ]
}
PS C:\WORK\Repos\webgme-cwl> node .\app.js
{
  enable: true,
  allowDuplication: true,
  defaultProject: 'cwl_base',
  basePaths: [
    'C:\\WORK\\Repos\\webgme-cwl\\config/../src/seeds/CWL',
    'C:\\WORK\\Repos\\webgme-cwl\\config/../node_modules/webgme-taxonomy/src/seeds/taxonomy'
  ],
  createAtStartup: []
}
2023-03-09T03:09:05.062Z - info: [gme] Node version v18.14.2
2023-03-09T03:09:05.063Z - info: [gme] Logfile [ info ] : C:\WORK\Repos\webgme-cwl\server.log
2023-03-09T03:09:05.063Z - info: [gme] Logfile [ error ] : C:\WORK\Repos\webgme-cwl\server-error.log
2023-03-09T03:09:05.081Z - info: [gme:server:standalone] Valid addresses of gme web server:  http://192.168.68.77:8888  http://127.0.0.1:8888  http://172.18.32.1:8888
Error: Tried loading "webgme-json-importer/build/JSONImporter.umd" at C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/build/JSONImporter.umd.js then tried node's require("C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd") and it failed with error: Error: Cannot find module 'C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd'
Require stack:
- C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js
- C:\WORK\Repos\webgme-cwl\node_modules\webgme\webgme.js
- C:\WORK\Repos\webgme-cwl\app.js
    at C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2600:27
    at Object.execCb (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1946:33)
    at Module.check (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1133:51)
    at Module.enable (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1426:22)
    at Module.init (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1038:26)
    at callGetModule (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1453:63)
    at Object.completeLoad (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1840:21)
    at req.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2613:17)
    at Object.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1935:21)
    at Module.load (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1084:29) {
  originalError: Error: Cannot find module 'C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\src/..\..\webgme-json-importer\src\common/./build/JSONImporter.umd'
  Require stack:
  - C:\WORK\Repos\webgme-cwl\node_modules\webgme-engine\index.js
  - C:\WORK\Repos\webgme-cwl\node_modules\webgme\webgme.js
  - C:\WORK\Repos\webgme-cwl\app.js
      at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
      at Module._load (node:internal/modules/cjs/loader:920:27)
      at Module.require (node:internal/modules/cjs/loader:1141:19)
      at require (node:internal/modules/cjs/helpers:110:18)
      at C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:2598:75
      at Object.execCb (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1946:33)
      at Module.check (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1133:51)
      at Module.enable (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1426:22)
      at Module.init (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1038:26)
      at callGetModule (C:\WORK\Repos\webgme-cwl\node_modules\requirejs\bin\r.js:1453:63) {
    code: 'MODULE_NOT_FOUND',
    requireStack: [
      'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme-engine\\index.js',
      'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme\\webgme.js',
      'C:\\WORK\\Repos\\webgme-cwl\\app.js'
    ]
  },
  moduleName: 'C:\\WORK\\Repos\\webgme-cwl\\node_modules\\webgme-engine\\src/..\\..\\webgme-json-importer\\src\\common/./build/JSONImporter.umd',
  requireModules: [ 'webgme-json-importer/build/JSONImporter.umd' ]
}
PS C:\WORK\Repos\webgme-cwl> npm list webgme-json-importer
webgme-cwl@1.0.0 C:\WORK\Repos\webgme-cwl
`-- webgme-taxonomy@1.2.1
  `-- webgme-json-importer@1.4.0 (git+ssh://git@github.com/deepforge-dev/webgme-json-importer.git#6ea33a9d29bec754014b1291e2a3f0d3ad803427)
'''
the result is the same no matter if webgme-json-importer is used in the cwl or just through taxonomy

--- a/package.json
+++ b/package.json
@@ -42,7 +42,7 @@
     "node-fetch": "^2.6.6",
     "svelte": "^3.50.0",
     "webgme": "github:webgme/webgme#azure",
-    "webgme-json-importer": "github:deepforge-dev/webgme-json-importer",
+    "webgme-json-importer": "^1.4.1",
     "zip-a-folder": "^1.1.5"
   },
   "lint-staged": {

